### PR TITLE
ebi-ait/dcp-ingest-central#382: updated nxn.se SOP

### PR DESCRIPTION
**Pull request by yusra-haider**
corresponding ticket: https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/382

related PR: https://github.com/ebi-ait/hca-ebi-dev-team/pull/445

### CODE REVIEW

**Review by ESapenaVentura — APPROVED** [review]
I really like it, very concise and clear

**Comment by ESapenaVentura** [comment]
When trying to run it:

## Errors

**Command**: `python3 populate_ingest/populate_ingest_from_nxn.py -tp "/Users/enrique/HumanCellAtlas/hca-ebi-dev-team/scripts/populate_ingest/token"` and `python3 populate_ingest_from_nxn.py -tp "/Users/enrique/HumanCellAtlas/hca-ebi-dev-team/scripts/populate_ingest/token"`

**Behaviour**:
```
Traceback (most recent call last):
  File "populate_ingest_from_nxn.py", line 14, in <module>
    from . import config
ImportError: attempted relative import with no known parent package
```

I cannot get it to work so for now I can't test it

File: docs/SOPs/dataset_acknowledgement_SOP.md [hunk 1]
```diff
@@ -34,48 +34,40 @@ At any point before halfway through the sprint, the wrangler/person responsible
...
-2) Run `python3 compare_tracker_with_nxn_sheet.py -c | pbcopy`
+2) Run `populate_ingest_from_nxn.py -tp token_path.txt`
+    
+   The script runs against local host, in dry-run mode by default
+   
+   The `-tp` parameter is for the txt file path with the authentication token.
+   
+   To run against another environment, specify the value of the environment variable `INGEST_API_URL`, 
+   for e.g. against the dev env:
+   `export INGEST_API_URL=https://dev.contribute.data.humancellatlas.org`
```
Comment (ESapenaVentura) [inline docs/SOPs/dataset_acknowledgement_SOP.md:56]: That's the UI URL, not the API :)
```suggestion
   `export INGEST_API_URL=https://api.ingest.dev.archive.data.humancellatlas.org/` 
```

File: docs/SOPs/dataset_acknowledgement_SOP.md [hunk 2]
```diff
@@ -34,48 +34,40 @@ At any point before halfway through the sprint, the wrangler/person responsible
...
-   cd hca-ebi-wrangler-central/src/
+   git clone https://github.com/ebi-ait/hca-ebi-dev-team.git
+   cd scripts/populate-ingest/
    pip install -r requirements.txt
    ```
 
 #### nxn.se single cell database
 
-1) Go to the `src/` folder inside your cloned version of the `hca-ebi-wrangler-central` repository
+1) Go to the `scripts/populate-ingest/` folder inside your cloned version of the `hca-ebi-dev-team` repository
```
Comment (amnonkhen) [inline docs/SOPs/dataset_acknowledgement_SOP.md:46]: Is this going to be manually run? Or will we connect it to some scheduler to be run periodically?

**Comment by ESapenaVentura** [comment]
I tried testing it today but I am still getting import errors (After pulling latest for the mouse change)

**Running it from python:**

```
python3 -m populate_ingest.populate_ingest_from_nxn -tp "/Users/enrique/HumanCellAtlas/hca-ebi-dev-team/scripts/populate_ingest/token"
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/enrique/HumanCellAtlas/hca-ebi-dev-team/scripts/populate_ingest/populate_ingest/populate_ingest_from_nxn.py", line 14, in <module>
    import config
ModuleNotFoundError: No module named 'config'
```

**Running it from terminal**
```
python3 -m populate_ingest.populate_ingest_from_nxn -tp "/Users/enrique/HumanCellAtlas/hca-ebi-dev-team/scripts/populate_ingest/token"
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/enrique/HumanCellAtlas/hca-ebi-dev-team/scripts/populate_ingest/populate_ingest/populate_ingest_from_nxn.py", line 14, in <module>
    import config
ModuleNotFoundError: No module named 'config'
```

File: docs/SOPs/dataset_acknowledgement_SOP.md (hunk 2 above)
Comment (yusra-haider) [reply docs/SOPs/dataset_acknowledgement_SOP.md]: manually, for now

File: docs/SOPs/dataset_acknowledgement_SOP.md [hunk 3]
```diff
@@ -34,48 +34,46 @@ At any point before halfway through the sprint, the wrangler/person responsible
 #### Requirements
 
 - Python3
-- Clone the `hca-ebi-wrangler-central` repo if you haven’t already and install the requirements:
+- Clone the `hca-ebi-dev-team` repo if you haven’t already and install the requirements:
    ```
-   git clone https://github.com/ebi-ait/hca-ebi-wrangler-central.git
-   cd hca-ebi-wrangler-central/src/
+   git clone https://github.com/ebi-ait/hca-ebi-dev-team.git
+   cd scripts/populate-ingest/
```
[the remark this answers is not in the document]
Comment (yusra-haider) [reply docs/SOPs/dataset_acknowledgement_SOP.md:40]: good catch!

**Review by jacobwindsor — APPROVED** [review]

File: docs/SOPs/dataset_acknowledgement_SOP.md [hunk 4]
```diff
@@ -34,48 +34,46 @@ At any point before halfway through the sprint, the wrangler/person responsible
...
+By default, the script runs against local host
 
+2) Run `python -m populate_ingest.populate_ingest_from_nxn`
+    
+   The script runs against local host, in dry-run mode by default
+   
+   To run in write mode, and write to ingest, use the `w` flag, e.g.
+   `python -m populate_ingest.populate_ingest_from_nxn.py -w`
+   
+   This script compares data between ingest and the nxn.se database, using `doi`, `accessions` and `title`, filters
```
[the remark this answers is not in the document]
Comment (yusra-haider) [reply docs/SOPs/dataset_acknowledgement_SOP.md]: moved

**Pull request merged**